### PR TITLE
Fresh 1176

### PR DESCRIPTION
**Pull request by metas-rc**

### CODE REVIEW

File: de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUQueryBuilder.java [hunk 1]
```diff
@@ -839,16 +881,36 @@ public IHUQueryBuilder addOnlyWithBPartnerLocationId(final Integer bpartnerLocat
 		return this;
 	}
 
+	/**
+	 * Kepp the attribute propagation as before.
```
Comment (metas-ts) [inline de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUQueryBuilder.java]: i don't understand this javadoc or the one from `getAttributeFilterVO()`

File: de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUQueryBuilder.java [hunk 2]
```diff
@@ -937,9 +999,34 @@ public boolean matches(final IAttributeSet attributes)
 	public IHUQueryBuilder setOnlyWithBarcode(final String barcode)
```
Comment (metas-ts) [inline de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUQueryBuilder.java]: please update the javadoc in IHUQueryBuilder to describe what the method is really achieving now

File: de.metas.handlingunits.client/src/test/java/de/metas/handlingunits/client/terminal/editor/view/HUKeysByBarcodeCollectorTest.java [hunk 3]
```diff
@@ -73,20 +72,20 @@ private void assertBarcodeMatches(final boolean matchedExpected, final HUKeysByB
 	public void init_Error_NullBarcode()
```
Comment (metas-ts) [inline de.metas.handlingunits.client/src/test/java/de/metas/handlingunits/client/terminal/editor/view/HUKeysByBarcodeCollectorTest.java]: please add unit tests to verify you new feature

File: de.metas.dimension/src/main/java/de/metas/dimension/IDimensionspecDAO.java [hunk 4]
```diff
@@ -73,4 +73,13 @@
...
 	List<String> retrieveAttributeValueForGroup(String dimensionSpectInternalName, String groupName, IContextAware ctxAware);
+
+	/**
+	 * @param internalName not empty or null
+	 * @param ctx
+	 * @param trxName
+	 * 
+	 * @return
+	 */
+	I_DIM_Dimension_Spec retrieveForInternalName(String internalName, Properties ctx, String trxName);
```
Comment (metas-rc) [inline de.metas.dimension/src/main/java/de/metas/dimension/IDimensionspecDAO.java]: First I thought that context was the reason of my stackoverflow in the console and I forgot to change it back and test after I found out it wasn't.
Fixed now.

File: de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUQueryBuilder.java (hunk 1 above)
Comment (metas-rc) [inline de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUQueryBuilder.java]: Fixed javadoc

File: de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUQueryBuilder.java (hunk 2 above)
Comment (metas-rc) [inline de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUQueryBuilder.java]: done

**Pull request merged**